docs/feature/policies/overview.md

```diff
@@ -0,0 +1,215 @@
+            # (...)
+    ```
+
+Engine will search for rules for a given Interface in the same order as specified in the list above. If an entry for a given Interface is found, then Engine uses it to fetch Implementations from Hub.
+
+For every Interface, Cluster Admin can set the order of selected Implementations, based on theirs constraints. The order of the list is important, as it is taken into account by Engine during queries to Hub. Engine iterates over list of `oneOf` items until it finds at least one Implementation satisfying the Implementation constraints.
+
+### Selecting Implementations
+
+You can select Implementations based on the following Implementation constraints:
+
+- `path`, which specifies the exact path for the Implementation. If the Implementation is found, then **any** revision of the Implementation is used. 
+
+    ```yaml
+    - interface:
+        path: cap.interface.database.postgresql.install
+      oneOf:
+        - implementationConstraints:
+            path: "cap.implementation.bitnami.postgresql.install" # any revision can be used
+    ```
+
+- `attributes`, which specifies the Attributes the selected Implementation must contain.
+
+    ```yaml
+    - interface:
+        path: cap.interface.database.postgresql.install
+      oneOf:
+        - implementationConstraints:
+            attributes:
+              - path: "cap.attribute.cloud.provider.gcp"
+                revision: "0.1.0"
+              - path: "cap.attribute.workload.stateful"
+                # any revision
+    ```
+
+- `requires`, which specifies the Type references, which should be included in the `spec.requires` field for an Implementation to be selected.
+
+    ```yaml
+    - interface:
+        path: cap.interface.database.postgresql.install
+      oneOf:
+        - implementationConstraints:
+            requires:
+              - path: "cap.core.type.platform.kubernetes" # any revision
+              - path: "cap.type.gcp.auth.service-account"
+                revision: "0.1.0" # exact revision 
+    ```
+
+- Empty constraints, which means any Implementation for a given Interface.
+    
+    ```yaml
+    - interface:
+        path: cap.interface.database.postgresql.install
+      oneOf:
+        - implementationConstraints: {} # any Implementation that implements the Interface
+    ```
+
+You can also deny all Implementations for a given Interface with the following syntax:
+
+```yaml
+- interface:
+    path: cap.interface.database.postgresql.install
+  oneOf: [] # deny all Implementations for a given Interface
+```
+
+### TypeInstance injection
+
+Along with Implementation constraints, Cluster Admin may configure TypeInstances, which are downloaded and injected in the Implementation workflow. This can be helpful for example, in case you are using Implementations, which are deploying infrastructure on a public cloud (like AWS or GCP) and you want to ensure that everything is deployed in the same account. For example:
```

Comment:
Maybe a note here with a link to the doc where we describe how to create such TypeInstance with credentials?